feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/presentation/handling/BaseNotificationHandler.kt

```diff
@@ -0,0 +1,35 @@
+package io.novafoundation.nova.feature_push_notifications.data.presentation.handling
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.os.Build
+import androidx.annotation.StringRes
+import androidx.core.app.NotificationManagerCompat
+import com.google.firebase.messaging.RemoteMessage
+import io.novafoundation.nova.common.resources.ResourceManager
+
+abstract class BaseNotificationHandler(
+    private val notificationManager: NotificationManagerCompat,
+    val resourceManager: ResourceManager,
+    @StringRes private val channelIdRes: Int,
+    @StringRes private val channelNameRes: Int,
+    private val importance: Int = NotificationManager.IMPORTANCE_DEFAULT
+) : NotificationHandler {
+
+    final override suspend fun handleNotification(message: RemoteMessage): Boolean {
+        val channelId = resourceManager.getString(channelIdRes)
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                channelId,
+                resourceManager.getString(channelNameRes),
+                importance
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+
+        return handleNotificationInternal(notificationManager, channelId, message)
+    }
+
+    protected abstract fun handleNotificationInternal(notificationManager: NotificationManagerCompat, channelId: String, message: RemoteMessage): Boolean
```

Comment:
Imho we can make `notificationManager` protected in the class and access it that way in subclasses instead